src/BowlingGame.js

```diff
@@ -0,0 +1,116 @@
+'use strict';
+
+var BowlingGame = function () {
+  this.LOWESTSCORE = 0;
+  this.HIGHESTSCORE = 270;
+  this.MAXFRAMES = 10;
+  this.score = 0;
+  this.noOfFrames = [];
+}
+
+BowlingGame.prototype.addFrame = function (frame) {
+  if (this._addFrameLimit() === true) {
+    return 'Game is over';
+  } else if (this._framesLengthIs9() === true) {
+    this._isFinalFrame(frame);
+  }  else {
+    this._incrementNoOfFrames(frame);
+    this._incrementScore();
+  }
+}
+
+BowlingGame.prototype.giveOutcome = function () {
+  if (this._isLowest() === true) {
+    return 'Gutter game! Too bad, try again next time!';
+  } else if (this._isHighest() === true) {
+    return 'Perfect game!';
+  } else {
+    return 'Your score is ' + this.score;
+  }
+}
+
+BowlingGame.prototype._addFrameLimit = function () {
+  return this.noOfFrames.length === this.MAXFRAMES;
+}
+
+BowlingGame.prototype._incrementNoOfFrames  = function (frame) {
+  this.noOfFrames.push(frame);
+}
+
+BowlingGame.prototype._incrementScore = function () {
+  if (this._lastTwoFramesWereStrikes() === true) {
+    this.score += ((this._currentFrame().rollsTotal) * 2) + (this._currentFrame().roll1);
```

Comment:
Could the Frame object have accessors for `rollsTotal`, `roll1` etc in order to achieve better encapsulation of state.